src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).get();
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        List<MovieSession> availableSessions = movieSessionDao.findAvailableSessions(movieId, date);
+        if (availableSessions.isEmpty()) {
+            throw new DataProcessingException("Available movie sessions by id = "
+                    + movieId + " and date: " + date + " are not exist!");
+        }
+        return availableSessions;
```

Comment:
Are you sure we need to drop an exception if the list is empty ?

Reply:
I think yes because user should see that something wrong and he should receive information about this.

Reply:
But when you drop an exception, you stop the programme running. An exception is suitable if there is an error in the program (wrong data, null, etc.). Here there is no error, you just did not find a session on a specific date. Because of this, dropping an exception is not a good idea.

Reply:
System.out.println?